src/Sphere.jl

```diff
@@ -35,3 +35,26 @@ end
 
 zero_tangent_vector(S::Sphere, x) = zero(x)
 zero_tangent_vector!(S::Sphere, v, x) = (v .= zero(x))
+
+function is_manifold_point(S::Sphere,x)
+    if size(x) != S.shape
+        throw(DomainError(size(x),"The point $(x) does not lie on $S, since its size is not $(S.shape)."))
+    end
+    if abs(norm(x)-1) > 10^(-15)
```

Comment:
Maybe we should provide tolerance keyword arguments like in `Base.isapprox`, which could just be forwarded to an `isapprox` call on this line.

Reply:
I tried, but I am still failing then providing a larger value of tolerance for the dot in the tangent test.

Reply:
It feels a bit clunky passing a whole approx function. I was thinking something more like.

```julia
function is_manifold_point(S::Sphere, x; kwargs...)
   ...
   if isapprox(norm(x), 1; kwargs...)
      ...
   end
end
```
This also makes it easier to change the tolerances. Instead of needing to call something like `is_manifold_point(m, x, x->isapprox(x; atol=1e-3))`, you'd call `is_manifold_point(m, x; atol=1e-3)`.